src/Vault.sol

```diff
@@ -422,30 +392,14 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver {
         uint256 vaultAmount = 0;
 
         uint256[] memory nfts = _vaultNfts[vaultId].values();
-        INonfungiblePositionManager positionManager_ = positionManager;
 
         for (uint256 i = 0; i < nfts.length; ++i) {
-            UniswapV3FeesCalculation.PositionInfo memory positionInfo;
-            uint256 nft = nfts[i];
-            (
-                ,
-                ,
-                ,
-                ,
-                ,
-                positionInfo.tickLower,
-                positionInfo.tickUpper,
-                positionInfo.liquidity,
-                positionInfo.feeGrowthInside0LastX128,
-                positionInfo.feeGrowthInside1LastX128,
-                positionInfo.tokensOwed0,
-                positionInfo.tokensOwed1
-            ) = positionManager_.positions(nft);
-            vaultAmount += _calculateAdjustedCollateral(_uniV3PositionInfo[nft], DENOMINATOR, positionInfo);
+            (, uint256 positionAmount, ) = oracle.price(nfts[i]);
```

Comment:
Each NFT's price is calculated twice, first in `calculateVaultAdjustedCollateral` and then here. Let's do some additional refactoring and calculate them only once.